### PR TITLE
types: finalize and cleanup types

### PR DESCRIPTION
**Pull request by itohatweb**

### CODE REVIEW

File: src/types/guilds/modify_guild_role.ts [hunk 1]
```diff
@@ -13,4 +15,7 @@ export interface ModifyGuildRole {
 }
 
 /** https://discord.com/developers/docs/resources/guild#modify-guild-role */
-export type DiscordModifyGuildRole = ModifyGuildRole;
+export interface DiscordModifyGuildRole
+  extends SnakeCaseProps<Omit<ModifyGuildRole, "permissions">> {
+  permissions?: string;
+}
```
Comment (ayntee) [inline src/types/guilds/modify_guild_role.ts:21]:
```suggestion
export interface DiscordModifyGuildRole
  extends Omit<ModifyGuildRole, "permissions"> {
  permissions?: string;
}
```

File: src/types/guilds/create_guild_role.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export interface CreateGuildRole {
 }
 
 /** https://discord.com/developers/docs/resources/guild#create-guild-role */
-export type DiscordCreateGuildRole = SnakeCaseProps<
-  CreateGuildRole
->;
+export interface DiscordCreateGuildRole
+  extends SnakeCaseProps<Omit<CreateGuildRole, "permissions">> {
```
Comment (ayntee) [inline src/types/guilds/create_guild_role.ts:19]:
```suggestion
  extends CreateGuildRole {
```

File: src/types/guilds/modify_guild_role.ts [hunk 3]
```diff
@@ -13,4 +14,7 @@ export interface ModifyGuildRole {
 }
 
 /** https://discord.com/developers/docs/resources/guild#modify-guild-role */
-export type DiscordModifyGuildRole = ModifyGuildRole;
+export interface DiscordModifyGuildRole
+  extends Omit<ModifyGuildRole, "permissions"> {
```
Comment (ayntee) [inline src/types/guilds/modify_guild_role.ts:18]:
```suggestion
  extends ModifyGuildRole {
```

File: src/types/guilds/create_guild_role.ts (hunk 2 above)
Comment (itohatweb) [reply src/types/guilds/create_guild_role.ts:19]: This does not work

File: src/types/guilds/modify_guild_role.ts (hunk 3 above)
Comment (itohatweb) [reply src/types/guilds/modify_guild_role.ts:18]: This does not work

File: src/types/guilds/create_guild_role.ts (hunk 2 above)
Comment (ayntee) [reply src/types/guilds/create_guild_role.ts:19]: It should work just fine

Comment (itohatweb) [reply src/types/guilds/create_guild_role.ts:19]: nope
```ts
Interface 'DiscordModifyGuildRole' incorrectly extends interface 'ModifyGuildRole'.
  Types of property 'permissions' are incompatible.
```

Comment (ayntee) [inline src/types/guilds/create_guild_role.ts:19]:
```suggestion
  extends Omit<CreateGuildRole, "permissions"> {
```

**Pull request merged**